src/ucan-cloud/ucan-gateway.ts

```diff
@@ -0,0 +1,367 @@
+    if (result.isErr()) this.logger.Error().Msg(result.Err().message);
+
+    const { store } = getStore(url, this.sthis, (...args) => args.join("/"));
+    if (result.isErr()) console.log("🚨", store, result.Err().constructor.name, result.Err().message);
+
+    return result;
+  }
+
+  async #put(url: URI, body: Uint8Array): Promise<void> {
+    const { store } = getStore(url, this.sthis, (...args) => args.join("/"));
+
+    if (this.inst === undefined) {
+      throw new Error("Not started yet");
+    }
+
+    const key = url.getParam("key");
+    if (!key) {
+      throw new Error("Key not found in the URI");
+    }
+
+    const name = url.getParam("name");
+    if (!name) {
+      throw new Error("Name not found in the URI");
+    }
+
+    console.log("🥘 PUT", store, key);
+
+    switch (store.toLowerCase()) {
+      case "data": {
+        await Client.store({
+          agent: this.inst.w3.agent.issuer,
+          bytes: body,
+          cid: CID.parse(key).toV1(),
+          server: this.inst.server,
+          service: this.inst.service,
+        });
+        break;
+      }
+
+      case "meta": {
+        const bodyWithCrypto = await bs.addCryptoKeyToGatewayMetaPayload(url, this.sthis, body);
+        if (bodyWithCrypto.isErr()) throw bodyWithCrypto.Err();
+        const metadata = bodyWithCrypto.Ok();
+
+        // const cid = CID.parse(key).toV1();
+        const event = await Client.createClockEvent({ metadata });
+
+        await Client.store({
+          agent: this.inst.w3.agent.issuer,
+          bytes: event.bytes,
+          cid: event.cid,
+          server: this.inst.server,
+          service: this.inst.service,
+        });
+
+        const { clock, server, service } = this.inst;
+        const agent = await this.agent();
+
+        const advancement = await Client.advanceClock({ agent, clock, event: event.cid, server, service });
+        if (advancement.out.error) throw advancement.out.error;
+
+        break;
+      }
+    }
+  }
+
+  async get(url: URI): Promise<bs.GetResult> {
+    const result = await exception2Result(() => this.#get(url));
+    if (result.isErr()) this.logger.Error().Msg(result.Err().message);
+
+    const { store } = getStore(url, this.sthis, (...args) => args.join("/"));
+    if (result.isErr()) console.log("🚨", store, result.Err().constructor.name, result.Err().message);
+
+    return result;
+  }
+
+  async #get(url: URI): Promise<Uint8Array> {
+    const { store } = getStore(url, this.sthis, (...args) => args.join("/"));
+
+    if (this.inst === undefined) {
+      throw new Error("Not started yet");
+    }
+
+    const key = url.getParam("key");
+    if (!key) {
+      throw new Error("Key not found in the URI");
+    }
+
+    let name = url.getParam("name");
+    if (!name) {
+      throw new Error("Name not found in the URI");
+    }
+
+    const index = url.getParam("index");
+    if (index) {
+      name += `-${index}`;
+    }
+
+    name += ".fp";
```

Comment:
there is a suffix in the url for car files and I think we should avoid implicit suffixes